src/app.ts

```diff
@@ -1,7 +1,24 @@
 import "./components/maincontroller";
 
-let senders = cast.framework.CastReceiverContext.getInstance().getSenders();
-let id = senders.length !== 0 && senders[0].id ? senders[0].id : new Date().getTime();
+declare global {
+    let PRODUCTION: boolean;
+    interface Window {
+        deviceInfo: deviceInfo;
+        mediaElement: HTMLElement;
+        playlist: Array<any>;
+        currentPlaylistIndex: number;
+        repeatMode: "RepeatOne" | "RepeatAll" | "RepeatNone";
+    }
+}
+
+interface deviceInfo {
```

Comment:
Best practice is generally to use PascalCase for interface names. It avoid confusing them with variables.